File: kiss.py
z = 1234567
w = 2345678
jsr = 3456789
jc = 4567890

def znew():
    global z
    z = 36969 * (z & 65535) + (0xffffffff & (z >> 16))
    return z

def wnew():
    global w
    w = 18000 * (w & 65535) + (0xffffffff & (w >> 16))
    return w
    
def MWC():
    return (0xffffffff & (znew() << 16)) + wnew()
    
def SHR3():
    global jsr
    jsr ^= (0xffffffff & (jsr << 17))
    jsr ^= (0xffffffff & (jsr >> 13))
    jsr ^= (0xffffffff & (jsr << 5))
    return jsr
    
def CONG():
    global jc
    jc = (0xffffffff & (69069 * jc)) + 1234567
    #jc = (0xffffffff & jc)
    return jc
    
def kiss():
    return ((MWC() ^ CONG()) + SHR3()) / 4294967296.0

f = open("datapizdata.txt", "w")
#c = 0

for i in range(10000):
        ##print(kiss() % 1)
        f.write(f"{round(kiss() % 1)}")
f.close()
        
#def kiss(z, w, jc, jsr):
    #z = 36969 * (z & 65535) + (0xffffffff & (z >> 16))
    #print(0xffffffff & (z << 16))
    #w = 18000 * (w & 65535) + (0xffffffff & (w >> 16))
    #jc = 69069 * jc + 1234567
    #jsr ^= (0xffffffff & (jsr << 17))
    #jsr ^= (0xffffffff & (jsr >> 13))
    #jsr ^= (0xffffffff & (jsr << 5))
    #return ((((0xffffffff & (z << 16)) + w) ^ jc) + jsr) / 4294967296.0

#for i in range(100):
    #print(kiss(1234567, 2345678, 3456789, 4567890))

File: kiss.c
#include <stdio.h>
#include <math.h>



__inline double rand01_kiss(void)
{
#define znew(z)   (z = 36969 * (z & 65535) + (z >> 16))
#define wnew(w)   (w = 18000 * (w & 65535) + (w >> 16))
#define MWC(z, w) ((znew(z) << 16) + wnew(w))
#define SHR3(jsr) (jsr ^= (jsr << 17), jsr ^= (jsr >> 13), jsr ^= (jsr << 5))
#define CONG(jc)  (jc = 69069 * jc + 1234567)
#define KISS(z, w, jc, jsr) ((MWC(z, w) ^ CONG(jc)) + SHR3(jsr))
  static unsigned z = 1234567, w = 2345678, jsr = 3456789, jcong = 4567890;
  return KISS(z, w, jcong, jsr) / 4294967296.0;
}



/* Gaussian distribution with zero mean and unit variance
 * using the ratio method */
__inline double gaussrand_kiss(void)
{
  double x, y, u, v, q;
  do {
    u = 1 - rand01_kiss();
    v = 1.7156*(rand01_kiss() - .5);  /* >= 2*sqrt(2/e) */
    x = u - 0.449871;
    y = fabs(v) + 0.386595;
    q = x*x  + y*(0.196*y - 0.25472*x);
    if (q < 0.27597) break;
  } while (q > 0.27846 || v*v > -4*u*u*log(u));
  return v/u;
}


int main(void)
{
  int i;

  for ( i = 0; i < 100; i++ )
    printf("%f\n", rand01_kiss());
  return 0;
}